sdks/java/core/src/main/java/org/apache/beam/sdk/util/MergingActiveWindowSet.java

```diff
@@ -160,99 +110,80 @@ public void persist() {
       // No change.
       return;
     }
-    // All NEW windows must have been accounted for.
-    for (Map.Entry<W, Set<W>> entry : activeWindowToStateAddressWindows.entrySet()) {
-      Preconditions.checkState(
-          !entry.getValue().isEmpty(), "Cannot persist NEW window %s", entry.getKey());
-    }
-    // Should be no EPHEMERAL windows.
-    Preconditions.checkState(
-        activeWindowToEphemeralWindows.isEmpty(), "Unexpected EPHEMERAL windows before persist");
-
     valueState.write(activeWindowToStateAddressWindows);
     // No need to update originalActiveWindowToStateAddressWindows since this object is about to
     // become garbage.
   }
 
   @Override
-  @Nullable
-  public W mergeResultWindow(W window) {
-    return windowToActiveWindow.get(window);
+  public Set<W> getActiveAndNewWindows() {
+    return activeWindowToStateAddressWindows.keySet();
   }
 
   @Override
-  public Set<W> getActiveWindows() {
-    return activeWindowToStateAddressWindows.keySet();
+  public boolean isActive(W window) {
+    Set<W> stateAddressWindows = activeWindowToStateAddressWindows.get(window);
+    return stateAddressWindows != null && !stateAddressWindows.isEmpty();
   }
 
   @Override
-  public boolean isActive(W window) {
+  public boolean isActiveOrNew(W window) {
     return activeWindowToStateAddressWindows.containsKey(window);
   }
 
   @Override
   public void ensureWindowExists(W window) {
-    if (!windowToActiveWindow.containsKey(window)) {
-      Preconditions.checkState(!activeWindowToStateAddressWindows.containsKey(window));
+    if (!activeWindowToStateAddressWindows.containsKey(window)) {
```

Comment:
it is odd that the keys of `activeWindowToStateAddressWindows` contains windows which are actually just NEW and are _not_ active. Just saying.